README.md

```diff
@@ -1,50 +1,107 @@
 
 ## Setup
 
 If you are using `pipenv` (recommended), `pipenv install` is sufficient to install Python prerequisites.
 
-[A Google Service account is necessary for use with `gspread`.](https://gspread.readthedocs.io/en/latest/oauth2.html#for-bots-using-service-account) Credentials for this account should be stored in `service_account.json`, in the same directory as the script.
+You also need:
+1. [A Google Service account is necessary for use with `gspread`.](https://gspread.readthedocs.io/en/latest/oauth2.html#for-bots-using-service-account) Credentials for this account should be stored in `service_account.json`, in the same directory as the script.
+1. [ffmpeg](https://www.ffmpeg.org/) must also be installed locally for this tool to work. 
+1. [firefox](https://www.mozilla.org/en-US/firefox/new/) and [geckodriver](https://github.com/mozilla/geckodriver/releases) on a path folder like `/usr/local/bin`. 
+1. [fonts-noto](https://fonts.google.com/noto) to deal with multiple unicode characters during selenium/geckodriver's screenshots: `sudo apt install fonts-noto -y`. 
+1. Internet Archive credentials can be retrieved from https://archive.org/account/s3.php.
+
+### Configuration file
+Configuration is done via a config.json file (see [example.config.json](example.config.json)) and some properties of that file can be overwritten via command line arguments. Here is the current result from running the `python auto_archive.py --help`:
+
+<details><summary><code>python auto_archive.py --help</code></summary>
+
+
+
+```js
+usage: auto_archive.py [-h] [--config CONFIG] [--storage {s3,local,gd}] [--sheet SHEET] [--header HEADER] [--s3-private] [--col-url URL] [--col-folder FOLDER] [--col-archive ARCHIVE] [--col-date DATE] [--col-status STATUS] [--col-thumbnail THUMBNAIL] [--col-thumbnail_index THUMBNAIL_INDEX] [--col-timestamp TIMESTAMP] [--col-title TITLE] [--col-duration DURATION] [--col-screenshot SCREENSHOT] [--col-hash HASH]
+
+Automatically archive social media posts, videos, and images from a Google Sheets document. The command line arguments will always override the configurations in the provided JSON config
+file (--config), only some high-level options are allowed via the command line and the JSON configuration file is the preferred method.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --config CONFIG       the filename of the JSON configuration file (defaults to 'config.json')
+  --storage {s3,local,gd}
+                        which storage to use [execution.storage in config.json]
+  --sheet SHEET         the name of the google sheets document [execution.sheet in config.json]
+  --header HEADER       1-based index for the header row [execution.header in config.json]
+  --s3-private          Store content without public access permission (only for storage=s3) [secrets.s3.private in config.json]
+  --col-url URL         the name of the column to READ url FROM (default='link')
```

Comment:
It should be indicated that all columns are optional _except_ for the URL and Status columns